### PR TITLE
feat: Use always search proxy

### PR DESCRIPTION
**Pull request by gabrielfs7**
Remove direct calls from ElasticSearch. Use Proxy instead

https://oat-sa.atlassian.net/browse/ADF-258

- [ ] https://github.com/oat-sa/tao-core/pull/2933
- [ ] https://github.com/oat-sa/extension-tao-publishing/pull/111
- [ ] https://github.com/oat-sa/lib-tao-elasticsearch/pull/45

### CODE REVIEW

File: model/Metadata/Listener/ClassDeletionListener.php [hunk 1]
```diff
@@ -42,14 +42,14 @@ public function listen($event): void
...
         try {
-            $this->getElasticSearch()->remove($event->getClass()->getUri());
+            $this->getSearchProxy()->remove($event->getClass()->getUri());
         } catch (Exception $exception) {
             $this->getLogger()->error($exception->getMessage());
         }
     }
 
-    private function getElasticSearch(): ElasticSearch
+    private function getSearchProxy(): SearchProxy
```
Comment (gabrielfs7) [inline model/Metadata/Listener/ClassDeletionListener.php:51]: Just a note to update tao-core version on composer before merge

- [ ] Update tao-core

**Review by boajer — APPROVED** [review]
LFGM, looking forward to see demo 

- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

**Review by chinnu-focaloid — APPROVED** [review]
- [X] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [X] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [X] Acceptance criteria are respected
- [X] Pull request title and description are meaningful

**Comment by AntonNeviarouski** [comment]
QA verified on http://aadvanced-search-integration.playground.kitchen.it.taocloud.org:48321/

**Comment by gabrielfs7** [comment]
Already done in the scope of another PR

**Pull request closed without merging**